ddtrace/llmobs/_integrations/bedrock.py

```diff
@@ -144,6 +152,116 @@ def _extract_output_message_for_converse(response: Dict[str, Any]):
             return default_content
         return get_messages_from_converse_content(role, content)
 
+    @staticmethod
+    def _extract_output_message_for_converse_stream(
+        streamed_body: List[Dict[str, Any]]
+    ) -> Tuple[List[Dict[str, Any]], Dict[str, str], Dict[str, int]]:
+        """
+        Extract output messages from streamed converse responses.
+
+        Converse stream response comes in chunks. The chunks we care about are:
+        - a message start/stop event, or
+        - a content block start/stop event (for tool calls only currently)
+        - a content block delta event (for chunks of text in a message or tool call arg)
+        - usage metric information
+
+        For more info, see bedrock converse response stream response syntax:
+        https://docs.aws.amazon.com/bedrock/latest/APIReference/API_runtime_ConverseStream.html#API_runtime_ConverseStream_ResponseSyntax
+        """
+        usage_metrics: Dict[str, int] = {}
+        metadata: Dict[str, str] = {}
+        messages: List[Dict[str, Any]] = []
+
+        text_content_blocks: Dict[int, str] = {}
+        tool_content_blocks: Dict[int, Dict[str, Any]] = {}
+
+        current_message: Optional[Dict[str, Any]] = None
+
+        def process_message_end(current_message: Dict[str, Any]):
```

Comment:
imo we should lift this function and pass `text_content_blocks` as an argument and return `messages` to make it a bit more explicit about how the state is being computed.